h/app.py

```diff
@@ -85,15 +85,15 @@ def includeme(config):
     # retrieving services bound to the request.
     config.include("pyramid_services")
 
-    # Configure the transaction manager to support retrying retryable
-    # exceptions, and generate a new transaction manager for each request.
     config.add_settings(
         {
-            "tm.attempts": 3,
-            "tm.manager_hook": lambda request: transaction.TransactionManager(),
+            "tm.manager_hook": lambda request: transaction.TransactionManager(
+                explicit=True
```

Comment:
`explicit=True` prevents a new transaction (and therefore a new sqlalchemy session and DB connection) from being implicitly created if something late in the request processing cycle tries to access the DB after pyramid_tm has already closed the transaction. Such implicitly-created would not be closed by pyramid_tm and would leak DB connections. `explicit=True` makes it crash rather than implicitly creating a new transaction. We don't currently have any code that will trigger this crash AFAIK, so this is just a precaution